samcli/lib/providers/sam_function_provider.py

```diff
@@ -387,40 +388,11 @@ def _parse_layer_info(
             # In the list of layers that is defined within a template, you can reference a LayerVersion resource.
             # When running locally, we need to follow that Ref so we can extract the local path to the layer code.
             if isinstance(layer, dict) and layer.get("Ref"):
-                layer_logical_id = cast(str, layer.get("Ref"))
-                layer_resource = stack.resources.get(layer_logical_id)
-                if not layer_resource or layer_resource.get("Type", "") not in (
-                    SamFunctionProvider.SERVERLESS_LAYER,
-                    SamFunctionProvider.LAMBDA_LAYER,
-                ):
-                    raise InvalidLayerReference()
-
-                layer_properties = layer_resource.get("Properties", {})
-                resource_type = layer_resource.get("Type")
-                compatible_runtimes = layer_properties.get("CompatibleRuntimes")
-                codeuri: Optional[str] = None
-
-                if resource_type == SamFunctionProvider.LAMBDA_LAYER:
-                    codeuri = SamFunctionProvider._extract_lambda_function_code(layer_properties, "Content")
-
-                if resource_type == SamFunctionProvider.SERVERLESS_LAYER:
-                    codeuri = SamFunctionProvider._extract_sam_function_codeuri(
-                        layer_logical_id, layer_properties, "ContentUri", ignore_code_extraction_warnings
-                    )
-
-                if codeuri and not use_raw_codeuri:
-                    LOG.debug("--base-dir is presented not, adjusting uri %s relative to %s", codeuri, stack.location)
-                    codeuri = SamLocalStackProvider.normalize_resource_path(stack.location, codeuri)
-
-                layers.append(
-                    LayerVersion(
-                        layer_logical_id,
-                        codeuri,
-                        compatible_runtimes,
-                        layer_resource.get("Metadata", None),
-                        stack_path=stack.stack_path,
-                    )
```

Comment:
explain: pylint complains about `_parse_layer_info()` being too complex, here I extract some logic to `_locate_layer_from_ref()`

Reply:
We shouldn't have to make these large (in terms of diff size) unrelated refactors to get around tooling, the tooling should work for us. Makes it difficult to review and focus on the relevant changes. Could the new code be refactored in a way that doesn't require moving existing code around?

Reply:
The commits were reorganized to isolate pure refactors. Thanks a lot @aahung, that's very helpful! 🙌